src/psyclone/domain/nemo/transformations/nemo_loop_fuse.py

```diff
@@ -110,27 +109,8 @@ def validate(self, node1, node2, options=None):
             if var_info1.is_read_only() and var_info2.is_read_only():
                 continue
 
-            # TODO #1213 - try to find symbol in case of 'wildcard' imports.
-            try:
-                # Find the symbol for this variable - the lookup function
-                # checks automatically in outer scopes.
-                symbol = symbol_table.lookup(var_name)
-                if isinstance(symbol, DataSymbol):
-                    is_array = symbol.is_array
-                else:
-                    # This typically indicates a symbol is used that we do
-                    # not have detailed information for, e.g. based on a
-                    # generic 'use some_mod' statement. In this case use
-                    # the information based on the access pattern, which
-                    # is at least better than having no information at all.
-                    is_array = var_info1[0].indices is not None
-            except KeyError:
-                # TODO #845: Once we have symbol tables, any variable should
-                # be in a symbol table, so we have to raise an exception here.
-                # We need to fall-back to the old-style test, since we do not
-                # have information in a symbol table. So check if the access
-                # information stored an index:
-                is_array = var_info1[0].indices is not None
+            symbol = symbol_table.lookup(var_name)
```

Comment:
I've opened #1270 for that. I think my original idea that it is useful to test both sources of information at once is actually wrong, in my next PR I am actually only relying on the access info (since I especially query for the loop variable, and if an expression is independent of the loop variable, it doesn't matter if it is an array expression or not). If this is true, I will remove this function completely as part of #257.

Reply:
Done (same for the dependency_tools, which have the same code ... see code duplication issue #1269)